cmd/shim/main.go

```diff
@@ -540,7 +560,17 @@ func internalEnv(name string) (string, bool) {
 func runWithNesting(ctx context.Context, cmd *exec.Cmd) error {
 	if _, found := internalEnv("_DAGGER_ENABLE_NESTING"); !found {
 		// no nesting; run as normal
-		return cmd.Run()
+		if err := cmd.Start(); err != nil {
```

Comment:
This approach looks good, we just also need to update the use of `cmd.Run` on line 611 with the same thing